x-pack/plugins/security_solution/public/actions/show_top_n/default/show_top_n.tsx

```diff
@@ -56,16 +56,17 @@ export const createShowTopNAction = ({
   });
 
   return {
-    id: ID,
-    type: ID,
+    id: ACTION_ID,
+    type: ACTION_ID,
     order,
     getIconType: (): string => ICON,
     getDisplayName: ({ field }) => SHOW_TOP(field.name),
     getDisplayNameTooltip: ({ field }) => SHOW_TOP(field.name),
     isCompatible: async ({ field }) =>
       isInSecurityApp(currentAppId) &&
       fieldHasCellActions(field.name) &&
-      !UNSUPPORTED_FIELD_TYPES.includes(field.type),
+      !UNSUPPORTED_FIELD_TYPES.includes(field.type) &&
+      !!field.aggregatable,
```

Comment:
Please beware of this issue for future topN chart: https://github.com/elastic/kibana/issues/150158
Lens only support esType: keyword, and `non-nested` field, could we add some additional checks for esTypes and nested field please? https://github.com/elastic/kibana/commit/11f18e873370b8f347d0055fca5609bba1896331

Reply:
The comment will be addressed in this issue: https://github.com/elastic/kibana/issues/150347